pennylane/templates/subroutines/qdrift.py

```diff
@@ -0,0 +1,248 @@
+            Pauli gates
+        time (int or float): The time of evolution, namely the parameter :math:`t` in :math:`e^{-iHt}`
+        n (int): An integer representing the number of exponentiated terms
+        seed (int): The seed for the random number generator
+
+    Raises:
+        TypeError: The ``hamiltonian`` is not of type :class:`~.Hamiltonian`, or :class:`~.Sum`
+        QuantumFunctionError: If the coefficients of ``hamiltonian`` are trainable and are used
+            in a differentiable workflow.
+
+    **Example**
+
+    .. code-block:: python3
+
+        coeffs = [0.25, 0.75]
+        ops = [qml.PauliX(0), qml.PauliZ(0)]
+        H = qml.dot(coeffs, ops)
+
+        dev = qml.device("default.qubit", wires=2)
+        @qml.qnode(dev)
+        def my_circ():
+            # Prepare some state
+            qml.Hadamard(0)
+
+            # Evolve according to H
+            qml.QDrift(H, time=1.2, n = 10)
+
+            # Measure some quantity
+            return qml.probs()
+
+    >>> my_circ()
+    [0.71061676 0.         0.28938324 0.        ]
+
+
+    .. details::
+        :title: Usage Details
+
+        We can also compute the gradient with respect to the evolution time:
+
+        .. code-block:: python3
+
+            dev = qml.device("default.qubit", wires=2)
+
+            @qml.qnode(dev)
+            def my_circ(time):
+                # Prepare H:
+                H = qml.dot([0.2, -0.1], [qml.PauliY(0), qml.PauliZ(1)])
+
+                # Prepare some state
+                qml.Hadamard(0)
+
+                # Evolve according to H
+                qml.QDrift(H, time, n=10, seed=10)
+
+                # Measure some quantity
+                return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+
+        >>> args = qnp.array([1.23])
+        >>> print(qml.grad(my_circ)(*tuple(args)))
+        0.27980654844422853
+    """
+
+    def __init__(self, hamiltonian, time, n=1, seed=None, decomposition=None, id=None):
+        r"""Initialize the QDrift class"""
+
+        if isinstance(hamiltonian, Hamiltonian):
+            coeffs, ops = hamiltonian.terms()
+
+        elif isinstance(hamiltonian, Sum):
+            coeffs, ops = [], []
+            for op in hamiltonian:
+                try:
+                    coeffs.append(op.scalar)
+                    ops.append(op.base)
+                except AttributeError:  # coefficient is 1.0
+                    coeffs.append(1.0)
+                    ops.append(op)
+
+        else:
+            raise TypeError(
+                f"The given operator must be a PennyLane ~.Hamiltonian or ~.Sum got {hamiltonian}"
+            )
+
+        if any(requires_grad(coeff) for coeff in coeffs):
+            raise qml.QuantumFunctionError(
+                "The QDrift template currently doesn't support differentiation through the "
+                "coefficients of the input Hamiltonian. Please instantiate the operation "
+                "using coefficents with `requires_grad` set to False."
```

Comment:
Do we need the last sentence? What is the coefficients are jax/tf/torch objects?

Reply:
Removed last sentence